Hunting Queries/MultipleDataSources/ApplicationGrantedEWSPermissions.yaml

```diff
@@ -0,0 +1,55 @@
+id: c7941212-4ff9-4d2d-b38d-54d78fa087cc
+name: Application Granted EWS Permissions
+description: |
+  'Finds AD applications granted permissions to read users mailboxes via Exchange Web Services (EWS). A threat actor could add these permissions to an application they control in order to gain persistent access to user's mail.
+  Review the applications granted these permissions to ensure they are required and were granted legitimately.'
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+  - connectorId: AzureActiveDirectoryIdentityProtection
+    dataTypes:
+      - SecurityAlert (IPC)
+tactics:
+  - Collection
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1078.004
+  - T1114.002
+query: |
+  AuditLogs
+  | where Category == "ApplicationManagement"
+  | where OperationName has "Add app role assignment to service principal"
+  | extend UA = tostring(AdditionalDetails[0].value)
+  | mv-expand TargetResources
+  | extend ModifiedProps = TargetResources.modifiedProperties
+  | mv-expand ModifiedProps
+  | where ModifiedProps.newValue has "Use Exchange Web Services with full access to all mailboxes"
+  | extend Action = ModifiedProps.newValue
+  | extend User = tolower(tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName))
+  | join AuditLogs on CorrelationId
+  | mv-expand TargetResources1
+  | mv-expand TargetResources1.modifiedProperties
+  | project-reorder TargetResources1_modifiedProperties
+  | extend displayName_ = tostring(TargetResources1_modifiedProperties.displayName)
+  | extend AppId = iff(tostring(TargetResources1_modifiedProperties.displayName) == "ServicePrincipal.AppId", tostring(TargetResources1_modifiedProperties.newValue), "")
+  | extend AppName = iff(tostring(TargetResources1_modifiedProperties.displayName) == "ServicePrincipal.DisplayName", tostring(TargetResources1_modifiedProperties.newValue), "")
+  | summarize make_set(AppName), make_set(AppId) by TimeGenerated, ActivityDisplayName, UA, User, Result, OperationName, tostring(InitiatedBy), bin(TimeGenerated, 1d), tostring(Action)
+  | where tostring(set_AppId) != '[""]'
+  | project-reorder TimeGenerated, User, set_AppName
+  | join kind=leftouter (SecurityAlert
+  | where ProviderName == "IPC"
+  | extend User = tolower(tostring(parse_json(ExtendedProperties).["User Account"]))
+  | summarize count(AlertName) by bin(TimeGenerated, 1d), User) on TimeGenerated, User
```

Comment:
count() doesn't take any arguments, is the goal here to count per AlertName?  I assume not based on the remainder of the query.  If not, then you can remove it.  ALA will call this out as an Error, even though it handles it.